### ISSUE TITLE
Arakoon went down, Unix.Unix_error(Unix.ENOENT, "open", "/mnt/hdd1/ovh/arakoon/global_nsm_2/tlogs/243.tlx")

### ISSUE THREAD

**Issue by jeroenmaelbrancke** [openvstorage/arakoon#133]
We tried to remote collapse the arakoon node and found out that the collapse exited with an error.

```
arakoon: main: backtrace:
arakoon: main: Raised at file "map.ml", line 122, characters 16-25
arakoon: main: Called from file "src/core/lwt.ml", line 161, characters 4-40
arakoon: main: remote_collapsing_failed: End_of_file
```

arakoon version: 1.9.9

Error from the arakoon log file:

```
2016-08-30 17:13:49 398254 +0200 - perf-roub-01 - 3318/0 - arakoon - 78744 - warning - Exception during first_i_of /mnt/hdd1/ovh/arakoon/global_nsm_2/tlogs/243.tlx: Unix.Unix_error(Unix.ENOENT, "open", "/mnt/hdd1/ovh/arakoon/global_nsm_2/
tlogs/243.tlx"); backtrace:; Raised by primitive operation at file "src/unix/lwt_unix.ml", line 241, characters 19-36
2016-08-30 17:13:49 398306 +0200 - perf-roub-01 - 3318/0 - arakoon - 78745 - fatal - Exception in fsm thread: Unix.Unix_error(Unix.ENOENT, "open", "/mnt/hdd1/ovh/arakoon/global_nsm_2/tlogs/243.tlx"); backtrace:; Raised by primitive operat
ion at file "src/unix/lwt_unix.ml", line 241, characters 19-36
2016-08-30 17:13:49 398382 +0200 - perf-roub-01 - 3318/0 - arakoon - 78746 - fatal - going down: Unix.Unix_error(Unix.ENOENT, "open", "/mnt/hdd1/ovh/arakoon/global_nsm_2/tlogs/243.tlx"); backtrace:; Raised at file "map.ml", line 122, char
acters 16-25; Called from file "src/core/lwt.ml", line 161, characters 4-40
2016-08-30 17:13:49 398398 +0200 - perf-roub-01 - 3318/0 - arakoon - 78747 - fatal - after pick
```

Node was unable to recover or boot up due to store ahead of tlogs:

```
2016-08-31 06:51:22 521276 +0200 - perf-roub-01 - 69945/0 - arakoon - 36 - info - Taking job from compression queue...
2016-08-31 06:51:22 521289 +0200 - perf-roub-01 - 69945/0 - arakoon - 37 - info - Creating local store at /mnt/hdd1/ovh/arakoon/global_nsm_2/db/Tqj1zMtIxTlijc5c.db (lcnum=16384) (ncnum=8192)
2016-08-31 06:51:22 521551 +0200 - perf-roub-01 - 69945/0 - arakoon - 38 - info - verify_n_catchup_store; too_far_i=0 current_i=0 si_o:Some ("4603368")
2016-08-31 06:51:22 521564 +0200 - perf-roub-01 - 69945/0 - arakoon - 39 - fatal - too_far_i:0, si_o:Some ("4603368") should not happen: tlogs have been removed?
2016-08-31 06:51:22 521573 +0200 - perf-roub-01 - 69945/0 - arakoon - 40 - fatal - [rc=40] Store ahead of tlogs: s_i=4603368, tlog_i=0
```

Complete log file:
[ovs-arakoon-global_nsm_2.log.1.gz](https://github.com/openvstorage/arakoon/files/446716/ovs-arakoon-global_nsm_2.log.1.gz)

**Comment by toolslive** [comment]
fix: #134

**Comment by toolslive** [comment]
will be in 1.9.11
